### ISSUE TITLE
'make distcheck' fails on master

### ISSUE THREAD

**Issue by elliefm** [cyrusimap/cyrus-imapd#2845]
This isn't a problem while we're not releasing tarballs from it, but it's a blocker for 3.2

It's currently failing the first time it runs configure out of the extracted tarball, due to pkg-config being unable to find openssl.  Which is weird, since I have openssl installed, and I have it installed in the usual place, not some weird spot, and the system pkg-config finds it okay (as does a normal build)...

**Comment by elliefm** [comment]
Currently bisecting to try and figure out when/why this broke, but it's a bit of a process

**Comment by elliefm** [comment]
Bisect tells me it started breaking at fea57c21f32febf0b1cda43734d8d2315c79b387, which feels incoherent, but that's what it says

... maybe it's a side-effect of that naked `export` that got snuck in at the end of Makefile.am?

If I remember right, the change made by this commit got backed out later and became the cyrus-timezones package instead, but maybe the `export` has stuck around...

**Comment by elliefm** [comment]
I think this will fix it, but there's another issue in the way, so I won't merge it until I know it is actually the right fix: https://github.com/cyrusimap/cyrus-imapd/compare/master...elliefm:v31/2845-fix-make-distcheck

**Comment by elliefm** [comment]
Our cunit/cunit.pl script is too clever for its own good, it re-implements a bunch of file-juggling stuff that should've just been make's responsibility, but doesn't consider VPATH builds, so of course it does it all badly!

This is biting us on master because the cunit tests became "auto-enabled if cunit is found" (and therefore distcheck ends up trying to build/run them), whereas on 3.0 they are "disabled unless explicitly requested" so distcheck never sees them.

**Comment by elliefm** [comment]
Fixes so far are merged but distcheck will still not succeed until I do something about the cunit setup (which might just be reverting the auto-enable change, if we need it in a hurry)

**Comment by elliefm** [comment]
Unit tests are no-longer auto-enabled if you have libcunit, so `make distcheck` doesn't choke on the broken cunit.pl anymore.  Fixing cunit.pl is #2849 

Found and fixed a few other bad bits, once the cunit.pl problem had been moved out of the way and I could see them!  So `make distcheck` works now.  I also found a way to tell it to build with all our optional components enabled, so it's now much better at detecting a broken distribution.
